### PR TITLE
Draft of learning objectives - Function development

### LINKED ISSUE #22: Draft of learning objectives for function development

### PR DESCRIPTION
**Pull request by AndersAskeland**
Closes #22

### CODE REVIEW

**Comment by AndersAskeland** [comment]
Just realised that we might need to include something in relation to non standard evaluation / tidy evaluation. What does everyone think? I don't believe NSE is ever taught in any of the other courses.

**Comment by lwjohnst86** [comment]
@AndersAskeland I think we should definitely teach that because they will encounter it if they do any package dev

File: sessions/function-development.qmd [hunk 1]
```diff
@@ -28,13 +28,14 @@ package, documenting them, and how to stop your functions with an error.
...
-**Session objectives**:
-
--   Convert a chunk of R code into a function
--   Include a function in an R package
--   Provide basic function documentation using Roxygen
--   Use functions from other packages in a function in your own package
--   Return an error message from a function
+**Learning objectives**:
+
+-   Convert your R code into a function
```
Comment (lwjohnst86) [inline sessions/function-development.qmd:33]:
```suggestion
-   Review and apply the workflow for converting code into a function.
```

File: sessions/function-development.qmd [hunk 2]
```diff
@@ -28,13 +28,14 @@ package, documenting them, and how to stop your functions with an error.
...
 
-**Session objectives**:
-
--   Convert a chunk of R code into a function
--   Include a function in an R package
--   Provide basic function documentation using Roxygen
--   Use functions from other packages in a function in your own package
--   Return an error message from a function
+**Learning objectives**:
+
```
Comment (lwjohnst86) [inline sessions/function-development.qmd:32]:
```suggestion

- Explain some software development and functional programming principles that underlie effective R package development
```

File: sessions/function-development.qmd [hunk 3]
```diff
@@ -28,13 +28,14 @@ package, documenting them, and how to stop your functions with an error.
...
+-   Convert your R code into a function
+-   Make your function modular and reusable
```
Comment (lwjohnst86) [inline sessions/function-development.qmd:34]:
```suggestion
-   Identify approaches and "code patterns" for writing your functions that ensure they are internally self-contained and reusable
```

File: sessions/function-development.qmd [hunk 4]
```diff
@@ -28,13 +28,14 @@ package, documenting them, and how to stop your functions with an error.
...
+-   Understand dependecy management and use functions from other
+    packages in your own package
```
Comment (lwjohnst86) [inline sessions/function-development.qmd:36]:
```suggestion
-   Explain what dependency means in the context of packages and how you formally manage dependencies when writing functions as part of a package
```

I'm not sure if maybe package dependency makes more sense in another session or here, but we'll keep it here for now.

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Great start! Here's some suggested changes 😁

**Review by AndersAskeland — COMMENTED** [review]
NA

**Review by AndersAskeland — COMMENTED** [review]
Test

**Review by AndersAskeland — COMMENTED** [review]
@lwjohnst86 Should be ready for merging.

**Review by lwjohnst86 — APPROVED** [review]
Nice! I made some small edits and will merge now 😁

**Pull request merged**